src/Factory/Occurrences.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Factory;
+
+use App\Entity\Occurrence;
+use App\Entity\Tag;
+use App\Model\Feed\FeedItemOccurrence;
+use App\Repository\OccurrenceRepository;
+
+final class Occurrences
+{
+    public function __construct(
+        private readonly OccurrenceRepository $occurrenceRepository
+    ) {
+    }
+
+    /**
+     * Create tag or find matching in the database.
+     *
+     * @param array<FeedItemOccurrence> $occurrences
+     *   The tag names to create/lockup in the database as strings
+     * @param ?int $eventId
+     *   The id of the event related to the occurrences
+     *
+     * @return iterable<Occurrence>
+     *   Yield tag entities from the database
+     */
+    public function createOrLookup(array $occurrences, ?int $eventId): iterable
+    {
+        foreach ($occurrences as $item) {
+            $occurrencesEntity = $this->occurrenceRepository->findOneBy(['start' => $item->start, 'end' => $item->end]);
+            if (!is_null($occurrencesEntity)) {
+                if ($occurrencesEntity->getEvent()?->getId() == $eventId) {
```

Comment:
Why don't you use `$eventId` in the call to `findOneBy`?

Reply:
Fixed